clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -263,6 +264,15 @@ public FSDataOutputStream append(Path path, int i, Progressable progressable) th
     public boolean rename(Path src, Path dst) throws IOException {
         ObjectLocation srcObjectLoc = pathToObjectLocation(src);
         ObjectLocation dstObjectLoc = pathToObjectLocation(dst);
+        if (srcObjectLoc.getPath().isEmpty()) {
+            LOG.error("rename: src {} is root directory", src);
```

Comment:
S3A does this?!  Weird, it looks as legitimate a rename as any.

Reply:
yes, https://github.com/apache/hadoop/blob/2960d83c255a00a549f8809882cd3b73a6266b6d/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java#L1498